docker/puppetdb/ssl-setup.sh

```diff
@@ -0,0 +1,454 @@
+# * Escapes any forward slashes and ampersands on the RHS for you
+#
+# $1 - regexp to match
+# $2 - line to replace
+# $3 - file to operate on
+#
+# Example:
+#
+#    replaceline "^$mysetting.*" "mysetting = myvalue" /etc/myconfig
+#
+function replaceline {
+  backupfile $3
+  tmp=$3.tmp.`date +%s`
+  sed "s/$1/$(echo $2 | sed -e 's/[\/&]/\\&/g')/g" $3 > $tmp
+  mv $tmp $3
+  chmod 644 $3
+}
+
+# This function comments out a line in a file, based on a regexp
+#
+# $1 = regexp to match
+# $2 = file to operate on
+#
+# Example:
+#
+#    commentline "^$mysetting.*" /etc/myconfig
+#
+function commentline {
+  backupfile $2
+  tmp=$2.tmp.`date +%s`
+  sed "/$1/ s/^/# /" $2 > $tmp
+  mv $tmp $2
+}
+
+# This function appends a line to a file
+#
+# $1 - line to append
+# $2 - file to operate on
+#
+# Example:
+#
+#    appendline "mysetting = myvalue" /etc/myconfig
+#
+function appendline {
+  backupfile $2
+  tmp=$2.tmp.`date +%s`
+  cat $2 > ${tmp}
+  echo $1 >> ${tmp}
+  mv ${tmp} $2
+}
+
+# This function copies the necessary PEM files from Puppet to the PuppetDB
+# SSL directory.
+#
+# This expects various environment variables to have already been set to work.
+function copy_pem_from_puppet {
+  # orig_files=($orig_ca_file $orig_private_file $orig_public_file)
+  orig_files="$orig_ca_file $orig_private_file $orig_public_file"
+  # for orig_file in "${orig_files[@]}"; do
+  for orig_file in $orig_files; do
+    if [ ! -e $orig_file ]; then
+      echo "Warning: Unable to find all puppet certificates to copy"
+      echo
+      echo "  This tool requires the following certificates to exist:"
+      echo
+      echo "  * $orig_ca_file"
+      echo "  * $orig_private_file"
+      echo "  * $orig_public_file"
+      echo
+      echo "  These files may be missing due to the fact that your host's Puppet"
+      echo "  certificates may not have been signed yet, probably due to the"
+      echo "  lack of a complete Puppet agent run. Try running puppet first, for"
+      echo "  example:"
+      echo
+      echo "      puppet agent --test"
+      echo
+      echo "  Afterwards re-run this tool then restart PuppetDB to complete the SSL"
+      echo "  setup:"
+      echo
+      echo "      ${ssl_command} -f"
+      exit 1
+    fi
+  done
+  rm -rf $ssl_dir
+  mkdir -p $ssl_dir
+  echo "Copying files: ${orig_ca_file}, ${orig_private_file} and ${orig_public_file} to ${ssl_dir}"
+  cp -pr $orig_ca_file $ca_file
+  cp -pr $orig_private_file $private_file
+  cp -pr $orig_public_file $public_file
+}
+
+########
+# MAIN #
+########
+
+# Gather command line options
+while getopts "ifh" opt;
+do
+  case $opt in
+    i)
+      interactive=true ;;
+    f)
+      force=true ;;
+    h)
+      usage ;;
+    *)
+      usage ;;
+  esac
+done
+
+${interactive:=false}
+${force:=false}
+
+# Deal with interactive setups differently to non-interactive
+if $interactive
+then
+    echo "interactive mode not yet implemented"
+    exit 1
```

Comment:
Disabled interactive mode here since refactoring to comma-separated strings wasn't as easy as it was in other places